src/plugins/intel_gpu/src/plugin/transformations/swiglu_fusion.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "swiglu_fusion.hpp"
+
+#include "intel_gpu/op/swiglu.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/swish.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+SwiGLUFusion::SwiGLUFusion() {
+    using namespace ov::pass::pattern;
+
+    // Detect SwiGLU decomposition pattern
+    // SwiGLU(Xw, Xv, beta) = (Xw * (1.0 + exp(-beta * Xw))) * Xv
+    auto data = any_input();
+
+    // VariadicSplit(X, axis, split_lengths) = Xw, Xv
+    auto const_axis = wrap_type<ov::op::v0::Constant>();
+    auto const_split_lengths = wrap_type<ov::op::v0::Constant>();
+    auto variadic_split = wrap_type<ov::op::v1::VariadicSplit>({data, const_axis, const_split_lengths});
+
+    // Swish(Xw) = Xw * (1.0 + exp(-beta * Xw))
+    auto swish = wrap_type<ov::op::v4::Swish>({variadic_split});
+
+    // Mul(Xw, Xv) = Swish(Xw) * Xv
+    auto mul = wrap_type<ov::op::v1::Multiply>({swish, variadic_split});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto input = pattern_map.at(data);
+
+        auto vsplit = std::dynamic_pointer_cast<ov::op::v1::VariadicSplit>(pattern_map.at(variadic_split).get_node_shared_ptr());
+        if (!vsplit)
+            return false;
+        auto vsplit_intput_pshape = vsplit->get_input_partial_shape(0);
+        auto last_dim = vsplit_intput_pshape.rank().get_length() - 1;
+        if (vsplit_intput_pshape[last_dim].is_dynamic())
+            return false;
+
+        auto axis = std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_map.at(const_axis).get_node_shared_ptr());
+        if (!axis)
+            return false;
+        bool valid_axis_const_values = ov::op::util::has_constant_value<int64_t>(axis, -1) ||
+                                       ov::op::util::has_constant_value<int64_t>(axis, last_dim);
```

Comment:
Could you move this part to predicate for constant `const_axis` ?
You may probably use [this](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/include/transformations/utils/utils.hpp#L270) helper

Reply:
It seems difficult to move it to predicate because when I use `INT_CONSTANT_WITH_PREDICATE` helper, I could compare `value` with fixed value such as -1, but I couldn't take last dim value from consumer(variadic split) and pass it as variable in expression.